compaction/compaction_manager.cc

```diff
@@ -526,12 +526,12 @@ future<compaction_manager::compaction_stats_opt> compaction_manager::perform_com
     co_return co_await perform_task(std::move(task_executor));
 }
 
-future<> compaction_manager::perform_major_compaction(table_state& t, tasks::task_info info) {
+future<> compaction_manager::perform_major_compaction(table_state& t, std::optional<tasks::task_info> info) {
     if (_state != state::enabled) {
         co_return;
     }
 
-    co_await perform_compaction<major_compaction_task_executor>(info ? std::make_optional(info) : std::nullopt, &t, info.id).discard_result();
+    co_await perform_compaction<major_compaction_task_executor>(info, &t, info.value_or(tasks::task_info{}).id).discard_result();
```

Comment:
BTW, if info has a bool operator why are passing an optional in the interface?
Maybe we should convert to optimized_optional<tasks::task_info>.... (as a follow up in any case)

Reply:
In perform_compaction empty task_info means something different than nullopt.

Empty task info means that the executor has no parent (this is a case for regular compaction), while nullopt indicates that a task was run from different context (like during a single function testing) and we do not want to keep it in task manager.

Reply:
There is an explaining comment above perform_compaction declaration.